addons/web/static/tests/core/commands/command_service.test.js

```diff
@@ -0,0 +1,1214 @@
+import { beforeEach, expect, getFixture, test } from "@odoo/hoot";
+import { manuallyDispatchProgrammaticEvent, press } from "@odoo/hoot-dom";
+import { Deferred, animationFrame } from "@odoo/hoot-mock";
+import { contains, makeMockEnv, mountWithCleanup, patchWithCleanup } from "../../web_test_helpers";
+
+import { Component, reactive, xml } from "@odoo/owl";
+
+import { useCommand } from "@web/core/commands/command_hook";
+import { registry } from "@web/core/registry";
+import { useActiveElement } from "@web/core/ui/ui_service";
+import { HotkeyCommandItem } from "@web/core/commands/default_providers";
+import { browser } from "@web/core/browser/browser";
+
+class TestComponent extends Component {
+    static template = xml`<div />`;
+    static props = ["*"];
+}
+
+class Parent extends Component {
+    static template = xml`
+      <t t-component="props.componentInfo.Component" t-if="props.componentInfo.Component" />
+    `;
+    static props = ["*"];
+}
+
+const commandProviderRegistry = registry.category("command_provider");
+const commandSetupRegistry = registry.category("command_setup");
+
+let env;
+beforeEach(async () => {
+    commandProviderRegistry.getEntries().forEach(([key]) => {
+        if (!["command", "data-hotkeys", "default"].includes(key)) {
+            commandProviderRegistry.remove(key);
```

Comment:
what are you removing here? You don't re-add them after...

Reply:
the original file also cleaned that registry. That suite is a generic one and only uses generic providers: not the mail's, not the menus'. The expect.toHaveCount are impacted if other providers are present